src/wms/src/main/java/org/geoserver/wms/CascadedLegendRequest.java

```diff
@@ -76,9 +80,18 @@ public GetLegendGraphicRequest getDelegate() {
     public JSONArray getCascadedJSONRules() {
         InputStream is = null;
         BufferedReader bufferedReader = null;
+
         try {
+            WMSLayerInfo wmsLayerInfo = (WMSLayerInfo) getLayerInfo().getResource();
+            StyleInfo defaultRemoteStyle = getLayerInfo().getDefaultStyle();
+            // when set to use whatever on remote WMS server
+            if (defaultRemoteStyle.getName().isEmpty() || defaultRemoteStyle.getLegend() == null)
```

Comment:
Hmm... so this empty has become some sort of hidden convention that is not documented anywhere?
What about having a StyleInfo constant instead, that refuses modifications, and that is clearly named to match what it's doing?

Reply:
The official documentation does not have it, but the one sent to user has it in the final report. Looks like we will require a small tutorial as part of documentation to explain this feature in detail.

StyleInfo constant, sure sounds like a plan, will do that

Reply:
@imranrajjad can you point me out to the documentation that describes the enw options for configuring a cascaded WMS layer? Thank you.

Reply:
@nmco I will add a new PR with complete documentation detailing the new functionality